components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+            addTestIdps(idPManagementDAO);
+
+            boolean availabilityResult = idPManagementDAO.isIdPAvailableForAuthenticatorProperty(authenticator,
+                    property, value, tenantId);
+            assertEquals(availabilityResult, isAvailable);
+        }
+    }
+
+    @Test(dataProvider = "isIdPAvailableForAuthenticatorPropertyData")
+    public void testIsIdPAvailableForAuthenticatorPropertyException(int tenantId, String authenticator, String property,
+                                                                    String value, boolean isAvailable)
+            throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+        }
+        assertThrows(IdentityProviderManagementException.class, () ->
+                idPManagementDAO.isIdPAvailableForAuthenticatorProperty(authenticator, property, value, tenantId));
+    }
+
+    @DataProvider
+    public Object[][] isIdpReferredBySPData() {
+
+        return new Object[][]{
+                {"testIdP1", SAMPLE_TENANT_ID, false},
+                {"testIdP2", SAMPLE_TENANT_ID, false},
+        };
+    }
+
+    @Test(dataProvider = "isIdpReferredBySPData")
+    public void testIsIdpReferredBySP(String idPName, int tenantId, boolean isAvailable) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            boolean availabilityResult = idPManagementDAO.isIdpReferredBySP(idPName, tenantId);
+            assertEquals(availabilityResult, isAvailable);
+        }
+    }
+
+    @Test(dataProvider = "isIdpReferredBySPData")
+    public void testIsIdpReferredBySPException(String idPName, int tenantId, boolean isAvailable) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+        }
+        assertThrows(IdentityProviderManagementException.class, () ->
+                idPManagementDAO.isIdpReferredBySP(idPName, tenantId));
```

Comment:
Add inside the try block.

Reply:
For having a exception this block should placed out the try block.